dashboard/api.py

```diff
@@ -592,5 +606,23 @@ def refresh_user_data(user_id):
         try:
             CachedEdxDataApi.update_cache_if_expired(user, edx_client, cache_type)
         except:
+            save_cache_update_failure(user_id)
             log.exception("Unable to refresh cache %s for student %s", cache_type, user.username)
             continue
+
+
+def save_cache_update_failure(user_id):
+    """
+    Store the number of time update cache failed for a user
+
+    Args:
+        user_id (int): The user id
+    """
+    con = get_redis_connection("redis")
+    user_key = FIELD_USER_ID_BASE_STR.format(user_id)
+    if con.hexists(CACHE_KEY_FAILURE_NUMS_BY_USER, user_key):
```

Comment:
It looks like this check isn't necessary. If the key doesn't exist `hincrby` assumes a starting value of 0 and adds to that: https://redis.io/commands/hincrby